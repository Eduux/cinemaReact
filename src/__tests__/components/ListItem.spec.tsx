import React from 'react';
import { render, fireEvent, getByText, act } from '@testing-library/react';

import Component from 'components/ListMovies/partials/ListItem';

import withFavorites, { initialState } from 'containers/favorites';

describe('ListItem component', () => {
  const mockMovie = {
    imdbID: '1',
    poster: '1',
    title: 'filme',
    type: 'a',
    year: '112',
  };

  it('should render', async () => {
    const Container: React.FC = withFavorites(Component.bind(null, mockMovie));

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must be clicked on the favorite button', () => {
    const Container: React.FC = withFavorites(
      Component.bind(null, mockMovie),
      initialState,
    );

    const result: any = render(<Container />);
    const { container } = result;

    act(() => {
      fireEvent(
        getByText(container, 'Favoritar'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });

    expect(result).toMatchSnapshot();
  });

  it('must be clicked on the unfavorite button', () => {
    const Container: React.FC = withFavorites(Component.bind(null, mockMovie), {
      list: [mockMovie],
    });

    const result: any = render(<Container />);
    const { container } = result;

    act(() => {
      fireEvent(
        getByText(container, 'Desfavoritar'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });

    expect(result).toMatchSnapshot();
  });
});
