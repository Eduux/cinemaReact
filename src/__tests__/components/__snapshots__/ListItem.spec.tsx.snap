// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component must be clicked on the favorite button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="sc-dkPtyc kIhNcZ"
      >
        <div
          class="sc-eCImvq dyuREx"
        >
          <img
            alt="filme"
            class="sc-bdvvaa cLlfML"
            height="auto"
            loading="lazy"
            src="1"
            width="auto"
          />
        </div>
        <div
          class="sc-hKwCoD fRRmAz"
        >
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            filme
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            a
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            112
          </strong>
          <div
            class="sc-gKckTs eMhRTy"
          >
            <button
              class="sc-gsDJrp ClvND"
              type="submit"
            >
              Desfavoritar
            </button>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="sc-dkPtyc kIhNcZ"
    >
      <div
        class="sc-eCImvq dyuREx"
      >
        <img
          alt="filme"
          class="sc-bdvvaa cLlfML"
          height="auto"
          loading="lazy"
          src="1"
          width="auto"
        />
      </div>
      <div
        class="sc-hKwCoD fRRmAz"
      >
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          filme
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          a
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          112
        </strong>
        <div
          class="sc-gKckTs eMhRTy"
        >
          <button
            class="sc-gsDJrp ClvND"
            type="submit"
          >
            Desfavoritar
          </button>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component must be clicked on the unfavorite button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="sc-dkPtyc kIhNcZ"
      >
        <div
          class="sc-eCImvq dyuREx"
        >
          <img
            alt="filme"
            class="sc-bdvvaa cLlfML"
            height="auto"
            loading="lazy"
            src="1"
            width="auto"
          />
        </div>
        <div
          class="sc-hKwCoD fRRmAz"
        >
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            filme
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            a
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            112
          </strong>
          <div
            class="sc-gKckTs eMhRTy"
          >
            <button
              class="sc-gsDJrp ClvND"
              type="submit"
            >
              Favoritar
            </button>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="sc-dkPtyc kIhNcZ"
    >
      <div
        class="sc-eCImvq dyuREx"
      >
        <img
          alt="filme"
          class="sc-bdvvaa cLlfML"
          height="auto"
          loading="lazy"
          src="1"
          width="auto"
        />
      </div>
      <div
        class="sc-hKwCoD fRRmAz"
      >
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          filme
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          a
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          112
        </strong>
        <div
          class="sc-gKckTs eMhRTy"
        >
          <button
            class="sc-gsDJrp ClvND"
            type="submit"
          >
            Favoritar
          </button>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="sc-dkPtyc kIhNcZ"
      >
        <div
          class="sc-eCImvq dyuREx"
        >
          <img
            alt="filme"
            class="sc-bdvvaa cLlfML"
            height="auto"
            loading="lazy"
            src="1"
            width="auto"
          />
        </div>
        <div
          class="sc-hKwCoD fRRmAz"
        >
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            filme
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            a
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            112
          </strong>
          <div
            class="sc-gKckTs eMhRTy"
          >
            <button
              class="sc-gsDJrp ClvND"
              type="submit"
            >
              Favoritar
            </button>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="sc-dkPtyc kIhNcZ"
    >
      <div
        class="sc-eCImvq dyuREx"
      >
        <img
          alt="filme"
          class="sc-bdvvaa cLlfML"
          height="auto"
          loading="lazy"
          src="1"
          width="auto"
        />
      </div>
      <div
        class="sc-hKwCoD fRRmAz"
      >
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          filme
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          a
        </strong>
        <strong
          class="sc-jRQAMF dxQwMq"
        >
          112
        </strong>
        <div
          class="sc-gKckTs eMhRTy"
        >
          <button
            class="sc-gsDJrp ClvND"
            type="submit"
          >
            Favoritar
          </button>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
